src/iceberg_extension.cpp

```diff
@@ -23,7 +23,18 @@
 
 namespace duckdb {
 
+bool AVRO_LOADED = false;
+
 static unique_ptr<BaseSecret> CreateCatalogSecretFunction(ClientContext &context, CreateSecretInput &input) {
+	if (!AVRO_LOADED) {
+		auto &db_instance = context.db;
+		ExtensionHelper::AutoLoadExtension(*db_instance, "avro");
```

Comment:
I don't have any opinions if this works, but I feel like I tried exactly this before and it didn't work.
AutoLoadExtension throws if it can't, the line below this will never be reached